src/rnn.py

```diff
@@ -15,12 +17,15 @@ def __init__(self, input_size, hidden_size, num_layers, batch_first):
         self.num_layers = num_layers
         # create the RNN cell
         self.batch_first = batch_first
+        self.dropoutLayer = nn.Dropout(p=dropout)
         self.cell = nn.GRU(
             self.input_size, self.hidden_size,
             self.num_layers, batch_first=self.batch_first)
         # print("Batch first: ", self.cell.batch_first)
 
     def forward(self, inputs, hidden, lengths=[], pad=True):
+        inputs = self.dropoutLayer(inputs)
```

Comment:
da valutare post training se questo dropout è di troppo, perchè tutti gli rnn ricevono in input delle frasi a cui è già stato applicato un dropout=0.1 alle parole, che vengono sostituite da <unk> (funzione noise).

Reply:
lei usa questo wrapper:
https://www.tensorflow.org/api_docs/python/tf/contrib/rnn/DropoutWrapper

dalle doc: "Operator adding dropout to inputs and outputs of the given cell."

però ok!